.changeset/selfish-coats-shout.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/config-loader': minor
```

Comment:
Wasn't sure if this should be major or minor as this _may be_ a fairly large breaking change.

Reply:
Either way we have to pick one path 😁 we can't both highlight it as breaking without also making it a `major` bump.

Tbh I don't think we need to treat this as breaking, because the impact this would have is really just that parts of config might unexpectedly be hidden. I might be wrong though, which is another reason to consider the alternate approach below.